### DISCUSSION TITLE
Adding custom filter logic when make()-ing columns

### DISCUSSION THREAD

**Discussion by fakeheal** [Power-Components/livewire-powergrid#859] [Ideas]
Hi,

we are using the package on models that have the `SoftDeletes` trait and it'd be useful to filter out soft deleted models from non-deleted ones. The column is `deleted_at` and is of type `datetime` and an input range doesn't really work for this, because you might miss out some records if you don't know how far in time you need to go back to select all records.

It'd be really cool if you could define your own logic (where clauses) when adding filters to columns:

```php
Column::make(...)
    ->makeBooleanFilter(dataField: 'deleted_at', ifActive: function ($query) {
        $query->whereNotNull('deleted_at');
    }, ifInactive: function ($query) {
        $query->whereNull('deleted_at');
    }),
```

I hope I didn't miss anything in the documentation or the discussions. Thanks!

**Comment by luanfreitasdev** [comment]
Hi @fakeheal , this is a good idea, however I intend to add this in version 4 along with the new way to use filters

https://v4.livewire-powergrid.com/table/column-filters.html#filter-boolean
